gnucash/gnome-utils/gnc-tree-view-account.c

```diff
@@ -256,6 +256,29 @@ gnc_tree_view_account_placeholder_toggled (GtkCellRendererToggle *cell,
     gtk_tree_path_free (s_path);
 }
 
+static void
+gnc_tree_view_account_opening_balance_toggled (GtkCellRendererToggle *cell,
+        const gchar *s_path_str,
+        gpointer user_data)
+{
+    GncTreeViewAccount *tree_view;
+    GtkTreePath *s_path;
+    Account *account;
+    gboolean opening_balance;
+
+    /* Change the requested account */
+    tree_view = user_data;
+    s_path = gtk_tree_path_new_from_string (s_path_str);
+    account = gnc_tree_view_account_get_account_from_path (tree_view, s_path);
+    if (account)
+    {
+        opening_balance = !gtk_cell_renderer_toggle_get_active (cell); // hasn't changed yet.
+        xaccAccountSetOpeningBalance (account, opening_balance);
+    }
+
+    /* Clean up */
+    gtk_tree_path_free (s_path);
+}
```

Comment:
> This lacks the controls of dialog-account that ensure that the account is of the right type, there are no splits in the account and that there isn't already an opening balance account for the account's commodity.

Because this check is already be performed in gnc_account_opening_balance_button_update(): would it not be enough to check if the toggle button is enabled and if not, return without any change instead of duplicating the checks ?

Reply:
Is that event handler really required ? It looks that it can be removed complety because setting a break point at  xaccAccountSetIsOpeningBalance() shows that it is only called on closing the dialog using the following call tree

1   xaccAccountSetIsOpeningBalance    Account.cpp      4173 0x7fffec140e4b 
2   gnc_ui_to_account                 dialog-account.c 470  0x7fffef8a123e 
3   gnc_finish_ok                     dialog-account.c 599  0x7fffef8a18d6 
4   gnc_edit_account_ok               dialog-account.c 913  0x7fffef8a2b00 
5   gnc_account_window_response_cb    dialog-account.c 989  0x7fffef8a2fe2 

while gnc_tree_view_account_opening_balance_toggled (), into which is also set a breakpoint, is never called while editing the account dialog.

Reply:
> Because this check is already be performed in gnc_account_opening_balance_button_update(): 

OK, mostly. That bails out if the account isn't an equity account, so if the user changes the account from equity to not-equity there's no way to make the check box not sensitive or `FALSE`.

> s that event handler really required ? 

Sorry, I've lost you. Which event handler?

Reply:
> Sorry, I've lost you. Which event handler?

Sorry for be'ing somehow unspecific - I refered to gnc_tree_view_account_opening_balance_toggled() which handles the button "toggle" event.

> s no way to make the check box not sensitiv

This is simple to solve by using something like this
```
static void
gnc_account_opening_balance_button_update (AccountWindow *aw, gnc_commodity *commodity)
{
    Account *account = aw_get_account (aw);
    Account *ob_account = gnc_account_lookup_by_opening_balance (gnc_book_get_root_account (aw->book), commodity);
    gboolean has_splits = xaccAccountCountSplits (account, FALSE) > 0;

    if (xaccAccountGetType (account) != ACCT_TYPE_EQUITY)
    {
        gtk_widget_set_sensitive (aw->opening_balance_button, FALSE);
        return;
    }
```